plaid/src/apis/npm/mod.rs

```diff
@@ -0,0 +1,115 @@
+mod hashes;
+pub mod npm_cli_client;
+pub mod npm_web_client;
+
+use std::sync::Arc;
+
+use regex::Regex;
+use reqwest::{cookie::Jar, Client};
+use serde::{Deserialize, Serialize};
+
+/// Client for interacting with npm that groups a client for CLI operations
+/// and one for web operations
+pub struct Npm {
+    config: NpmConfig,
+    client: Client,
+    cookie_jar: Arc<Jar>,
```

Comment:
@obelisk TBC if we need a `Mutex` here

Reply:
If it works without it then I think ti's fine. Rust should prevent you from doing something dangerous so this item must have interior mutability.